Android.bp

```diff
@@ -22,8 +42,16 @@ mfx_cc_defaults {
 
         "-DANDROID",
         "-DMFX_C2_USE_PRIME",
+        "-DLIBVA_SUPPORT",
+        "-DLIBVA_ANDROID_SUPPORT",
     ],
 
+    product_variables: {
+        platform_sdk_version: {
+            cflags: ["-DPLATFORM_SDK_VERSION=%d"],
+        },
```

Comment:
I see lots of variables are deleted in mfx_c2_defs.mk and mfx_c2_env.mk, such as MFX_VERSION and MFX_ANDROID_VERSION. Can we keep these variables to make sure we are compatible with older android version and hw platforms? Also, we need to make sure codec2.0 can also be compatible with mediasdk API.

Reply:
We can only keep these variables in .h file with preprocessing. Because Android.bp doesn't support if condition. I'll think about it.

Reply:
> We can only keep these variables in .h file with preprocessing. Because Android.bp doesn't support if condition. I'll think about it.

Please consider to use XXX.go statement to control/process the if condition. There is an example here. http://inlubt0232.iind.intel.com:8080/source/xref/CELADON/vendor/intel/mediasdk_c2/Android.bp?r=cb50fd88#72

Reply:
yes, it's a little bit complicated, so I just keep the variables that we have used in source code.
I've investigated about how to get the android version in android.bp, the sdk_version is only I can get, and I found it is also used in framework. So I think it will work. If you still need to keep these, I will consider to use go lang to implement it. Because I think some of them are unnecessary.

Reply:
from offical doc https://android.googlesource.com/platform/build/soong/+/HEAD/docs/best_practices.md#removing-conditionals

Using go lang is as the last resort, if you have to keep the if condition, I think as google, they don't suggest do it.